File: core/src/main/java/com/ustadmobile/core/view/ReportNumberOfDaysClassesOpenView.java
package com.ustadmobile.core.view;


import java.util.LinkedHashMap;

/**
 * ReportNumberOfDaysClassesOpen Core View extends Core UstadView. Will be implemented
 * on various implementations.
 */
public interface ReportNumberOfDaysClassesOpenView extends UstadView {

    String VIEW_NAME = "ReportNumberOfDaysClassesOpen";

    //Report Bar Chart colors
    String BAR_CHART_BAR_COLOR = "#004586";

    //Label:
    String BAR_CHART_LABEL_X = "Day";
    String BAR_CHART_LABEL_Y = "# classes open";
    String BAR_LABEL = "Number of classes open";

    //Attendance chart limits
    int BAR_CHART_HEIGHT = 200;

    /**
     * Takes a map of x, y values for the chart and updates the Line chart on the View.
     * @param dataMap The hashmap containing the values
     */
    void updateBarChart(LinkedHashMap<Float, Float> dataMap);


    /**
     * For Android: closes the activity.
     */
    void finish();

    /**
     * Generate csv report from the View which has the table data
     */
    void generateCSVReport();

    void generateXLSXReport(String path);

}

File: core/src/commonMain/kotlin/com/ustadmobile/core/view/ReportNumberOfDaysClassesOpenView.kt
package com.ustadmobile.core.view


/**
 * ReportNumberOfDaysClassesOpen Core View extends Core UstadView. Will be implemented
 * on various implementations.
 */
interface ReportNumberOfDaysClassesOpenView : UstadView {

    /**
     * Takes a map of x, y values for the chart and updates the Line chart on the View.
     * @param dataMap The hashmap containing the values
     */
    fun updateBarChart(dataMap: LinkedHashMap<Float, Float>)


    /**
     * For Android: closes the activity.
     */
    fun finish()

    /**
     * Generate csv report from the View which has the table data
     */
    fun generateCSVReport()

    fun generateXLSXReport(path: String)

    companion object {

        val VIEW_NAME = "ReportNumberOfDaysClassesOpen"

        //Report Bar Chart colors
        val BAR_CHART_BAR_COLOR = "#004586"

        //Label:
        val BAR_CHART_LABEL_X = "Day"
        val BAR_CHART_LABEL_Y = "# classes open"
        val BAR_LABEL = "Number of classes open"

        //Attendance chart limits
        val BAR_CHART_HEIGHT = 200
    }

}
